lcUI/lcadviewer.cpp

```diff
@@ -16,33 +21,145 @@ LCADViewer::LCADViewer(QWidget *parent) :
     _scaleLineWidth(false),
     _backgroundPainter(nullptr),
     _documentPainter(nullptr),
-    _foregroundPainter(nullptr) {
+    _foregroundPainter(nullptr)
+     {
+
 
+   
+    QOpenGLContext *CC= QOpenGLContext::currentContext();
+    
     setMouseTracking(true);
     this->_altKeyActive = false;
     this->_ctrlKeyActive = false;
-	setCursor(Qt::BlankCursor);
+    setCursor(Qt::BlankCursor);
+
+   QSurfaceFormat format;
+    format.setMajorVersion(4);
+    format.setMinorVersion(6);
+    format.setProfile(QSurfaceFormat::CoreProfile);
+    format.setOption(QSurfaceFormat::DebugContext);
+    setFormat(format);
+}
+
+void LCADViewer::messageLogged(const QOpenGLDebugMessage &msg)
+{
+    qCritical() << "OpenGL log:\n";
```

Comment:
You forgot a Qt log
This seems to be a callback of Qt logging. Use LibreCAD 3's logger in that case

Reply:
Do i need to delete this ?? This was for OpenGL debugging, I have'nt deleted these functions but disabled them.